docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/FeatureSelection/SelectFeaturesBasedOnMutualInformationMultiColumn.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    public static class SelectFeaturesBasedOnMutualInformationMultiColumn
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert it to an IDataView.
+            var rawData = GetData();
+
+            Console.WriteLine("Contents of columns 'Label', 'NumericVectorA' and 'NumericVectorB'.");
+            foreach (var item in rawData)
+                Console.WriteLine("{0}\t\t{1}\t\t{2}", item.Label, string.Join(" ", item.NumericVectorA), string.Join(" ", item.NumericVectorB));
+            // True            4 0 6           7 8 9
+            // False           0 5 7           7 9 0
+            // True            4 0 6           7 8 9
+            // False           0 5 7           7 8 0
+
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // We define a MutualInformationFeatureSelectingEstimator that selects the top k slots in a feature 
+            // vector based on highest mutual information between that slot and a specified label. 
+
+            // Multi column example : This pipeline transform two columns using the provided parameters.
+            var pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnMutualInformation(
+                new InputOutputColumnPair[] { new InputOutputColumnPair("NumericVectorA"), new InputOutputColumnPair("NumericVectorB") },
+                labelColumnName: "Label",
+                slotsInOutput: 4);
+
+            var transformedData = pipeline.Fit(data).Transform(data);
+
+            var convertedData = mlContext.Data.CreateEnumerable<TransformedData>(transformedData, true);
+            Console.WriteLine("Contents of two columns 'NumericVectorA' and 'NumericVectorB'.");
+            foreach (var item in convertedData)
+                Console.WriteLine("{0}\t\t{1}", string.Join(" ", item.NumericVectorA), string.Join(" ", item.NumericVectorB));
+
+            // Here, we see SelectFeaturesBasedOnMutualInformation selected 4 slots.
```

Comment:
>4 slot [](start = 76, length = 6)

the 4 slots that had most in common with the respective value in the Label column, maybe?

Reply:
yes, these 4 slots carried the most MI with Label....  we should have a better tutorial for this though

---
In reply to: [272398839](https://github.com/dotnet/machinelearning/pull/3184#discussion_r272398839) [](ancestors = 272398839)